### PR TITLE
Fix a grabbag of semi-automatically found typos throughout the codebase

### PR DESCRIPTION
**Pull request by cristicbz**
I found these automatically, but fixed them manually to ensure the semantics are correct. I know things like these are hardly important, since they only marginally improve clarity. But at least for me typos and simple grammatical errors trigger an---unjustified---sense of unprofessionalism, despite the fact that I make them all the time and I understand that they're the sort of thing that is bound to slip through review.  

Anyway, to find most of these I used:
- `ag '.*//.*(\b[A-Za-z]{2,}\b) \1\b'` for repeated words
- `ag '\b(the|this|those|these|a|it) (a|the|this|those|these|it)\b'` to find constructs like 'the this' etc. many false positives, but not too hard to scroll through them to actually find the mistakes.
- `cat ../../typos.txt | paste -d'|' - - - - - - - - - - - - - - - - - - - - - - | tr '\n' '\0' | xargs -0 -P4 -n1 ag`. Hacky way to find misspellings, but it works ok. I got `typos.txt` from [Wikipedia](https://en.wikipedia.org/wiki/Wikipedia:Lists_of_common_misspellings/For_machines)
- `ag '.*//.* a ([ae][a-z]|(o[^n])|(i[a-rt-z]))'` to find places where 'a' was followed by a vowel (requiring 'an' instead).

I also used a handful more one off regexes that are too boring to reproduce here.

### CODE REVIEW

File: src/librustdoc/html/toc.rs [hunk 1]
```diff
@@ -51,12 +51,12 @@ pub struct TocEntry {
 #[derive(PartialEq)]
 pub struct TocBuilder {
     top_level: Toc,
-    /// The current hierarchy of parent headings, the levels are
+    /// In the current hierarchy of parent headings, the levels are
```
Comment (steveklabnik) [inline src/librustdoc/html/toc.rs]: This change is bad

File: src/libstd/sys/windows/mutex.rs [hunk 2]
```diff
@@ -22,7 +22,7 @@
 //!    more details.
 //!
 //! 3. While CriticalSection is fair and SRWLock is not, the current Rust policy
-//!    is there there are no guarantees of fairness.
+//!    is there are no guarantees of fairness.
```
Comment (steveklabnik) [inline src/libstd/sys/windows/mutex.rs]: This should actually be "the current Rust policiy is that there are no guarantees of fairness."

**Comment by steveklabnik** [comment]
Thanks so much! This is awesome.

**Comment by cristicbz** [comment]
@steveklabnik Thanks for the review! Should I amend the commit, or add a new one and squash?

**Comment by steveklabnik** [comment]
Either way works :)

**Comment by cristicbz** [comment]
Done!

**Comment by steveklabnik** [comment]
not gonna roll this one up since it touches many files, might conflict

**Comment by cristicbz** [comment]
Someone else [picked up](https://github.com/steveklabnik/rust/commit/4119fc27498dba51d0233f0496ff8d03036ee350#diff-678c2da97c0ef7453998bf2f22434376R302) on one of 'the the'-s. I'll rebase tonight.

**Comment by steveklabnik** [comment]
Just ping me after you do, github doesn't send notifications for just rebases

**Comment by cristicbz** [comment]
Rebased!

**Comment by alexcrichton** [comment]
@bors: r+ 4b308b44e1d8204702f6912dda4dfc404aa2a87d

Thanks!

**Pull request merged**